docs/user/vsphere/customization.md

```diff
@@ -12,7 +12,11 @@ Beyond the [platform-agnostic `install-config.yaml` properties](../customization
 
 ## Machine pools
 
-There are currently no configurable vSphere-specific machine-pool properties.
+* `osDisk` (optional object):
+    * `diskSizeGB` (optional integer): The size of the disk in gigabytes (GB).
+* `numCPUs` (optional integer): The total number of virtual processor cores to assign a vm.
+* `numCoresPerSocket` (optional integer): The number of cores per socket in a vm. The number of vCPUs on the vm will be NumCPUs/NumCoresPerSocket (default is 1).
```

Comment:
Who would want to change this value from default 1??

Reply:
@abhinavdahiya customers may have software that is licensed per socket or core - we should account for that.
It is also defined in the MachineApi